plugins/src/gcc/imp.rs

```diff
@@ -0,0 +1,1270 @@
+                     encoder should also use it as target bitrate",
+                    1,
+                    u32::MAX as u32,
+                    DEFAULT_MIN_BITRATE as u32,
+                    glib::ParamFlags::READWRITE | gst::PARAM_FLAG_MUTABLE_READY,
+                ),
+                glib::ParamSpecUInt::new(
+                    "min-bitrate",
+                    "Minimal Bitrate",
+                    "Minimal bitrate to use (in bit/sec) when computing it through the bandwidth estimation algorithm",
+                    1,
+                    u32::MAX as u32,
+                    DEFAULT_MIN_BITRATE,
+                    glib::ParamFlags::READWRITE | gst::PARAM_FLAG_MUTABLE_READY,
+                ),
+                glib::ParamSpecUInt::new(
+                    "max-bitrate",
+                    "Minimal Bitrate",
+                    "Minimal bitrate to use (in bit/sec) when computing it through the bandwidth estimation algorithm",
+                    1,
+                    u32::MAX as u32,
+                    DEFAULT_MAX_BITRATE,
+                    glib::ParamFlags::READWRITE | gst::PARAM_FLAG_MUTABLE_READY,
+                ),
+            ]
+        });
+
+        PROPERTIES.as_ref()
+    }
+
+    fn set_property(
+        &self,
+        _obj: &Self::Type,
+        _id: usize,
+        value: &glib::Value,
+        pspec: &glib::ParamSpec,
+    ) {
+        match pspec.name() {
+            "min-bitrate" => {
+                let mut state = self.state.lock().unwrap();
+                state.min_bitrate = value.get::<u32>().expect("type checked upstream");
+            }
+            "max-bitrate" => {
+                let mut state = self.state.lock().unwrap();
+                state.max_bitrate = value.get::<u32>().expect("type checked upstream");
+            }
+            "estimated-bitrate" => {
+                let mut state = self.state.lock().unwrap();
+                let bitrate = value.get::<u32>().expect("type checked upstream");
+                state.target_bitrate_on_delay = bitrate;
+                state.target_bitrate_on_loss = bitrate;
+                state.estimated_bitrate = bitrate;
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn property(&self, _obj: &Self::Type, _id: usize, pspec: &glib::ParamSpec) -> glib::Value {
+        match pspec.name() {
+            "min-bitrate" => {
+                let state = self.state.lock().unwrap();
+                state.min_bitrate.to_value()
+            }
+            "max-bitrate" => {
+                let state = self.state.lock().unwrap();
+                state.max_bitrate.to_value()
+            }
+            "estimated-bitrate" => {
+                let state = self.state.lock().unwrap();
+                state.estimated_bitrate.to_value()
+            }
+            _ => unimplemented!(),
+        }
+    }
+}
+
+impl GstObjectImpl for BandwidthEstimator {}
+
+impl Default for State {
```

Comment:
This seems like a strange place between all the other trait impls for `BandwidthEstimator`. Maybe move it up to the definition of the `State` struct